shared-bindings/microcontroller/Processor.c

```diff
@@ -61,6 +72,27 @@
 //|     frequency: int
 //|     """The CPU operating frequency in Hertz. (read-only)"""
 //|
+
+STATIC mp_obj_t mcu_processor_set_sys_clock(mp_obj_t self_in, mp_obj_t freq) {
+    mcu_processor_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    #if defined(HAS_SETTABLE_CLOCK)
+    uint32_t value_of_freq = MP_OBJ_SMALL_INT_VALUE(freq);
```

Comment:
This does not validate it's an int that you passed in, and that it's >= 0. Use:
```suggestion
    uint32_t value_of_freq = (uint32_t)mp_arg_validate_min(0, mp_obj_get_int(freq), MP_QSTR_frequency);
```

Reply:
Also test this by hand after you write it to make sure it's OK; I may not be completely right here.

Reply:
Done